app/Http/Controllers/ProductController.php

```diff
@@ -11,15 +11,20 @@
 
 class ProductController extends Controller
 {
+    private $isPaginate = false;
 
     public function index(ProductRequest $request)
     {
-        $isPaginate = false;
-
         try{
             $query_res = Product::when($request->filled('startDate') && $request->filled('endDate'), function ($query) use ($request) {
                 return $query->whereBetween('date', [$request->startDate, $request->endDate]);
             })
+            ->when($request->start_date, function ($query, $start_date) {
+                return $query->where('date', '>=', $start_date);
+            })
+            ->when($request->end_date, function ($query, $end_date) {
+                return $query->where('date', '<=', $end_date);
+            })
```

Comment:
Stick to a consistent naming convention for variables. For example, use either startDate or start_date consistently

Reply:
done